server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutes.java

```diff
@@ -192,10 +238,37 @@ private User extractUser(Request request) {
         }
     }
 
+    private HaltException badRequest(String message) {
```

Comment:
these `badRequest` method make things more complicated than needed. You have a nice builder, just return it instead of creating various methods with similar arguments that requires to read the method body anyway.